src/coreclr/jit/flowgraph.cpp

```diff
@@ -3934,7 +3928,8 @@ PhaseStatus Compiler::fgSetBlockOrder()
             {
                 case BBJ_COND:
                 case BBJ_ALWAYS:
-                    partiallyInterruptible = EDGE_IS_GC_SAFE(block, block->GetJumpDest());
+                    partiallyInterruptible =
+                        ((block->bbFlags & BBF_NONE_QUIRK) != 0) || EDGE_IS_GC_SAFE(block, block->GetJumpDest());
```

Comment:
This is preexisting to your PR, but the logic here is wrong. It seems this logic is relying on `bbNum`'s to be increasing, but the fact that you had to add this quirk means that they aren't. But if they aren't then we aren't properly detecting potential loops here that need to be fully interruptible.
For example, consider this loop:
```
BB01 [BBJ_NONE]
BB00 [BBJ_ALWAYS -> BB01]
```
We do not look at the `BB01 -> BB00` edge at all, and the `BB00 -> BB01` edge is considered GC safe because of increasing BB nums.

Reply:
Thanks for catching this. I wonder if the TP regression we solved by adding the quirk here was an instance like the example you gave, where we do need fully-interruptible GC reporting. Since replacing `BBJ_NONE` with `BBJ_ALWAYS` to the next block, maybe the prior approach of always checking the jump destination is the right move?